NewRelicBrowser.php

```diff
@@ -1,392 +1,456 @@
 		}
 
 		/**
-		 * Create a browser app
-		 * @param string $app_name name of the app
-		 * @param string $account_api_key api key of the account
+		 * Create a browser app.
+		 *
+		 * @param string $app_name name of the app.
+		 * @param string $account_api_key api key of the account.
+		 *
 		 * @return boolean
 		 */
-		function rtp_create_browser_app( $app_name, $account_api_key )
-		{
+		public function rtp_create_browser_app( $app_name, $account_api_key ) {
 
 			$app_data = array(
 				browser_application => array(
 					'name' => $app_name,
-				)
+				),
 			);
-			$app_dataString = json_encode( $app_data );
+
+			$app_datastring = wp_json_encode( $app_data );
 
 			/* send request to create a browser app */
 
 			$app_response = wp_remote_post(
-				RTP_NEW_RELIC_BROWSER_URL, array(
-				'timeout' => 100,
-				'method' => 'POST',
-				'headers' => array(
-					'x-api-key' => $account_api_key,
-					'Content-Type' => 'application/json',
-				),
-				'body' => $app_dataString,
-					) );
+				RTP_NEW_RELIC_BROWSER_URL,
+				array(
+					'timeout' => 100,
+					'method'  => 'POST',
+					'headers' => array(
+						'x-api-key'    => $account_api_key,
+						'Content-Type' => 'application/json',
+					),
+					'body'    => $app_datastring,
+				)
+			);
 
-					/* get the response code */
+			/* get the response code */
 
-					$app_response_code = wp_remote_retrieve_response_code( $app_response );
+			$app_response_code = wp_remote_retrieve_response_code( $app_response );
 
-					$app_json_data = json_decode( $app_response['body'] );
-				if ( 201 !== $app_response_code ) {
-						add_settings_error( 'relic_options', 'relic_options_error', $app_json_data->error->title, 'error' );
-						return false;
-				} else {
-						/* stored the received browser application data */
-
-						$browser_details_array = array(
-							'relic_app_name' => $app_json_data->browser_application->name,
-							'relic_app_key' => $app_json_data->browser_application->browser_monitoring_key,
-							'relic_app_id' => $app_json_data->browser_application->id,
-							'relic_app_script' => $app_json_data->browser_application->loader_script,
-						);
-						add_option( 'rtp_relic_browser_details', $browser_details_array );
-						return true;
-				}
+			$app_json_data = json_decode( $app_response['body'] );
+			if ( 201 !== $app_response_code ) {
+					add_settings_error( 'relic_options', 'relic_options_error', $app_json_data->error->title, 'error' );
+					return false;
+			} else {
+					/* stored the received browser application data */
+
+					$browser_details_array = array(
+						'relic_app_name'   => $app_json_data->browser_application->name,
+						'relic_app_key'    => $app_json_data->browser_application->browser_monitoring_key,
+						'relic_app_id'     => $app_json_data->browser_application->id,
+						'relic_app_script' => $app_json_data->browser_application->loader_script,
+					);
+					add_option( 'rtp_relic_browser_details', $browser_details_array );
```

Comment:
Use `update_option()`

Reply:
i solved this issue. please review changes.